tests/core/utilities/test_method_formatters.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from web3._utils.method_formatters import (
+    get_error_formatters,
+    raise_if_revert_reason,
+)
+from web3._utils.rpc_abi import (
+    RPC,
+)
+from web3.exceptions import (
+    SolidityError,
+)
+from web3.types import (
+    RPCResponse,
+)
+
+REVERT_WITH_MSG = RPCResponse({
+    'jsonrpc': '2.0',
+    'error': {
+        'code': -32015,
+        'message': 'VM execution error.',
+        'data': (
+            'Reverted '
+            '0x08c379a'
+            '00000000000000000000000000000000000000000000000000000000000000020'
+            '0000000000000000000000000000000000000000000000000000000000000016'
+            '6e6f7420616c6c6f77656420746f206d6f6e69746f7200000000000000000000'
+        ),
+    },
+    'id': 2987,
+})
+
+REVERT_WITHOUT_MSG = RPCResponse({
+    'jsonrpc': '2.0',
+    'error': {
+        'code': -32015,
+        'message': 'VM execution error.',
+        'data': 'Reverted 0x',
+    },
+    'id': 2987,
+})
+
+OTHER_ERROR = RPCResponse({
+    "jsonrpc": "2.0",
+    "error": {
+        "code": -32601,
+        "message": "Method not found",
+    },
+    "id": 1,
+})
+
+
+@pytest.mark.parametrize(
+    "response,expected",
+    (
+        (REVERT_WITH_MSG, 'execution reverted: not allowed to monitor'),
+        (REVERT_WITHOUT_MSG, ''),
+    ),
+    ids=[
+        'test-get-revert-reason-with-msg',
+        'test-get-revert-reason-without-msg',
+    ])
+def test_get_revert_reason(response, expected) -> None:
+    with pytest.raises(SolidityError):
```

Comment:
This is pretty nitpicky, but that assert will never get hit. I think this would be better like:
```
    with pytest.raises(SolidityError, match=expected):
        raise_if_revert_reason(response)
```